src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.utils.tsx

```diff
@@ -58,13 +62,75 @@ export const getResponseItem = ({
     />
   );
 
+  const renderSingleSelectionPerRow = () => {
```

Comment:
Please use react component instead to use internal reconciliation mechanism to reduce a re-render on a side effect